
Reported upstream - https://github.com/tcalmant/jsonrpclib/pull/44

File "/usr/lib/python3.9/vendor-packages/jsonrpclib/jsonrpc.py", line 570, in __init__
 |     schema, uri = splittype(uri)
 |   File "/usr/lib/python3.9/urllib/parse.py", line 1009, in splittype
 |     warnings.warn("urllib.parse.splittype() is deprecated as of 3.8, "
 | DeprecationWarning: urllib.parse.splittype() is deprecated as of 3.8, use urllib.parse.urlparse() instead

--- jsonrpclib-0.4.1/jsonrpclib/jsonrpc.py~	2020-10-09 10:27:04.506390034 +0000
+++ jsonrpclib-0.4.1/jsonrpclib/jsonrpc.py	2020-10-09 10:32:34.471676543 +0000
@@ -69,7 +69,7 @@
     # Python 3
     # pylint: disable=F0401,E0611
     from http.client import HTTPConnection
-    from urllib.parse import splittype, splithost
+    from urllib.parse import urlparse
     from xmlrpc.client import Transport as XMLTransport
     from xmlrpc.client import SafeTransport as XMLSafeTransport
     from xmlrpc.client import ServerProxy as XMLServerProxy
@@ -567,7 +567,15 @@
         self._config = config
         self.__version = version or config.version
 
-        schema, uri = splittype(uri)
+        if sys.version_info[0] < 3:
+            schema, uri = splittype(uri)
+            self.__host, self.__handler = splithost(uri)
+        else:
+            su = urlparse(uri)
+            schema = su.scheme
+            self.__host = su.netloc
+            self.__handler = su.path
+
         use_unix = False
         if schema.startswith("unix+"):
             schema = schema[len("unix+"):]
@@ -578,7 +586,6 @@
                           schema)
             raise IOError('Unsupported JSON-RPC protocol.')
 
-        self.__host, self.__handler = splithost(uri)
         if use_unix:
             unix_path = self.__handler
             self.__handler = '/'
